src/main/41-lb.tf

```diff
@@ -0,0 +1,50 @@
+resource "aws_lb" "alb_eks" {
+  name               = "${var.app_name}-alb-${var.env}"
+  internal           = true
+  load_balancer_type = "application"
+  subnets            = data.aws_subnets.workload.ids
+
+  enable_deletion_protection       = true
+  enable_cross_zone_load_balancing = true
+  tags = {
+    "elbv2.k8s.aws/cluster"    = module.eks.cluster_name
+    "ingress.k8s.aws/resource" = "LoadBalancer"
+    "ingress.k8s.aws/stack"    = var.alb_ingress_group
+  }
+}
+
+
+resource "aws_lb" "nlb" {
+  name               = "${var.app_name}-nlb-${var.env}"
+  internal           = true
+  load_balancer_type = "network"
+  subnets            = data.aws_subnets.workload.ids
+
+  enable_deletion_protection       = true
+  enable_cross_zone_load_balancing = true
+}
+
+resource "aws_lb_listener" "front_end" {
+  load_balancer_arn = aws_lb.nlb.arn
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.alb.arn
+  }
+}
+
+resource "aws_lb_target_group" "alb" {
```

Comment:
Missing health check configuration, I suggest to match 200-499 status code range

Reply:
this is the NLB to ALB target group is it needed?

Reply:
Not strictly needed, but if you omit this the target will appear as unhealthy even if it's not true